src/Extracting/Generator.php

```diff
@@ -313,7 +313,14 @@ public static function setObject(array &$results, string $path, $value, array $s
             $baseNameInOriginalParams = substr($baseNameInOriginalParams, 0, -2);
         }
 
-        if (Arr::has($source, $baseNameInOriginalParams)) {
+        if (empty($baseNameInOriginalParams)) {
+            // This indicates that the body is an array of objects, so each parameter should be a key in the first object in that array
+            $results[0][$paramName] = $value;
+        } elseif (array_key_exists(0, $results)) {
+            // The body is an array of objects, so every parameter must be an element of the array object.
```

Comment:
Good question! The second scenario is one where the body is known to be an array, as evidenced by the existence of an element at key zero. In that case, since the body cannot be both an array and an object, we have to add any other named body parameters as being an attribute of the first object in the array.

I'll submit a commit shortly to make that comment clearer.